skimage/io/hdr.py

```diff
@@ -0,0 +1,291 @@
+import numpy as np
+
+
+def getRadiance(ims, exp, radianceMap):
+    """
+    Return the radiance for a series of images based upon a camera response 
+    function
+
+    Parameters
+    ----------
+    ims : list
+          list of images in the for of numpy arrays. Either mono or colour
+          (RGB, MxNx3)
+
+    exp : numpy 1D array
+          array of exposure times in seconds
+
+    radiacenMap : numpy array
+                  array mapping the counts to radiance
+
+    Returns
+    ----------
+    hdr : numpy array
+          resulting image with radiance values
+    """
+
+    den = np.ones(ims[0].shape)
+    num = np.zeros(ims[0].shape)
+    wf = np.vectorized(weightFunc)
+    for idx, im in enumerate(ims):
+        gij = im.copy()
+        # For colour
+        if im.ndim == 3:
+            for ii in range(im.shape[2]):
+                gij[:, :, ii] = radianceMap[im[:, :, ii] + 1, ii]
+        else:
+            gij[:, :] = radianceMap[im[:, :, ii] + 1]
+        gij = gij - np.log(exp[idx])
+
+        wij = wf(zij)
+        num = num + wij * gij
+        den = den + wij
+
+    return num / den
+
+
+def makeHdr(ims, exp, radianceMap, depth=16):
+    """
+    Compute the HDR image from a series of images and a racianceMap
+    Based on:
+    Debevec, P. E., & Malik, J. (1997). SIGGRAPH 97 Conf. Proc., August, 3–8.
+    doi:10.1145/258734.258884
+
+    Parameters
+    ----------
+    ims : list
+          list of images in the for of numpy arrays. Either grayscale or colour
+          (RGB)
+
+    exp : numpy 1D array
+          array of exposure times in seconds
+
+    radianceMap : numpy array
+               array (idx) mapping counts to radiance value, if input is RGB 
+               this must be Nx3 
+
+    depth : int, optional
+            pixel depth, default=16
+
+    Returns
+    ----------
+    hdr : numpy array
+          The HDR image either grayscale or RGB depending on input in ln(E)
+    """
+    B = np.log(np.array(exp))
+
+    if ims[0].ndim == 3:
+        sx, sy, sc = np.shape(ims[0])
+        hdr = np.zeros([sx, sy, sc], dtype=np.float)
+        gray = False
+    else:
+        sx, sy = np.shape(ims[0])
+        hdr = np.zeros([sx, sy], dtype=np.float)
+        gray = True
+
+    ims = np.asarray(ims, dtype=np.int)
+
+    sx, sy, sz = np.shape(ims[0])
+
+    w = np.vectorize(weightFunc)
```

Comment:
Please use `wf` for this variable as well, to make it clear that this is a function.